### PR TITLE
Fix: support for union type for custom form methods

### PR DESCRIPTION
**Pull request by lulco**
Follow up for https://github.com/efabrica-team/phpstan-latte/pull/223

I had to fix some issues

### CODE REVIEW

File: src/LatteContext/Collector/FormFieldCollector.php [hunk 1]
```diff
@@ -103,26 +104,39 @@ public function collectData(Node $node, Scope $scope): ?array
             }
 
             $formFieldType = $formFieldParametersAcceptor->getReturnType();
-
             $fieldNameArg = $node->getArgs()[0] ?? null;
-            $fieldNameDefaultType = $formFieldParametersAcceptor->getParameters()[0]->getDefaultValue();
+
+            $formFieldParameters = $formFieldParametersAcceptor->getParameters();
+            $fieldNameDefaultType = isset($formFieldParameters[0]) ? $formFieldParameters[0]->getDefaultValue() : null;
```
Comment (lulco) [inline src/LatteContext/Collector/FormFieldCollector.php:110]: here I got error "Call method getDefaultValue() on null"

File: src/LatteContext/Collector/FormFieldCollector.php [hunk 2]
```diff
@@ -103,26 +104,39 @@ public function collectData(Node $node, Scope $scope): ?array
...
             if ($fieldNameDefaultType instanceof ConstantStringType) {
                 $fieldNameDefault = trim($fieldNameDefaultType->getValue(), '"\'');
             } else {
                 $fieldNameDefault = null;
             }
         }
 
-        if (!$formFieldType instanceof ObjectType) {
-            return null;
+        $formFieldTypes = [];
```
Comment (lulco) [inline src/LatteContext/Collector/FormFieldCollector.php:118]: My method addAccount() has union return type, so I'm trying to solve it here. It works like expceted, I just don't like the code. Any idea how to write it more phpstan-like?

Comment (MartinMystikJonas) [reply src/LatteContext/Collector/FormFieldCollector.php:118]: Try TypeCombinator::intersect return type with union of allowed types.

**Review by MartinMystikJonas — APPROVED** [review]

File: src/LatteContext/Collector/FormFieldCollector.php (hunk 2 above)
Comment (lulco) [reply src/LatteContext/Collector/FormFieldCollector.php:118]: I'll play with it later, now I'm going to merge

**Pull request merged**